whois-nrtm4-client/src/main/java/net/ripe/db/nrtm4/client/importer/SnapshotImporter.java

```diff
@@ -0,0 +1,180 @@
+package net.ripe.db.nrtm4.client.importer;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.google.common.base.Stopwatch;
+import com.google.common.collect.Maps;
+import net.ripe.db.nrtm4.client.client.NrtmRestClient;
+import net.ripe.db.nrtm4.client.client.UpdateNotificationFileResponse;
+import net.ripe.db.nrtm4.client.condition.Nrtm4ClientCondition;
+import net.ripe.db.nrtm4.client.dao.Nrtm4ClientInfoRepository;
+import net.ripe.db.nrtm4.client.dao.Nrtm4ClientRepository;
+import net.ripe.db.whois.common.dao.RpslObjectUpdateInfo;
+import net.ripe.db.whois.common.rpsl.RpslObject;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.stereotype.Service;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+import static org.apache.commons.codec.binary.Hex.encodeHexString;
+
+@Service
+@Conditional(Nrtm4ClientCondition.class)
+public class SnapshotImporter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SnapshotImporter.class);
+
+    private final NrtmRestClient nrtmRestClient;
+
+    private final Nrtm4ClientInfoRepository nrtm4ClientInfoMirrorDao;
+
+    private final Nrtm4ClientRepository nrtm4ClientRepository;
+
+
+    public SnapshotImporter(final NrtmRestClient nrtmRestClient,
+                            final Nrtm4ClientInfoRepository nrtm4ClientInfoMirrorDao,
+                            final Nrtm4ClientRepository nrtm4ClientRepository) {
+        this.nrtmRestClient = nrtmRestClient;
+        this.nrtm4ClientInfoMirrorDao = nrtm4ClientInfoMirrorDao;
+        this.nrtm4ClientRepository  = nrtm4ClientRepository;
+    }
+
+    public void truncateTables(){
+        nrtm4ClientInfoMirrorDao.truncateTables();
+        nrtm4ClientRepository.truncateTables();
+    }
+
+    public Map<RpslObject, RpslObjectUpdateInfo> importSnapshot(final String source, final UpdateNotificationFileResponse updateNotificationFile){
+        final Stopwatch stopwatch = Stopwatch.createStarted();
+        final UpdateNotificationFileResponse.NrtmFileLink snapshot = updateNotificationFile.getSnapshot();
+
+        if (snapshot == null){
+            LOGGER.error("Snapshot cannot be null in the notification file");
+            return Maps.newHashMap();
```

Comment:
should this fail? i.e. throw an exception

Reply:
well, it is more or less the same that we talked in the office today?
We will retry this in a minute and this will be retried one again in a minute